tests/Entity/JsonPointerTest.php

```diff
@@ -92,4 +92,16 @@ public function getTestData()
 
         );
     }
+
+    public function testJsonPointerWithPropertyPaths()
+    {
+        $initial = new JsonPointer('#/definitions/date');
+        $modified = $initial->withPropertyPaths(array('~definitions/general', '%custom%'));
```

Comment:
To prove **withPropertyPaths** is immutable. The code needs to be changed here into:

    $this->assertEquals(array('definitions', 'date'), $initial->getPropertyPaths());
    $this->assertEquals('#/definitions/date', $initial->getPropertyPathAsString());
    $modified = $initial->withPropertyPaths(array('~definitions/general', '%custom%'));
    $this->assertEquals(array('definitions', 'date'), $initial->getPropertyPaths());
    $this->assertEquals('#/definitions/date', $initial->getPropertyPathAsString());